commit e1eabdde91b8d658edca71596fde56a3bad33cc1
Author: TensorFlow Hub Authors <no-reply@google.com>
Date:   Sun Sep 9 17:21:05 2018 +0200

    Fix "TypeError: Cannot mix str and non-str arguments" error when loading a module from https.
    
    Users were likely to have encountered this in google cloud datalab and composer.
    
    This is likely the fix for https://github.com/tensorflow/hub/issues/76.
    
    Tested: manually in google cloud datalab.
    PiperOrigin-RevId: 212179432

diff --git a/tensorflow_hub/compressed_module_resolver.py b/tensorflow_hub/compressed_module_resolver.py
index 14ad4c2..a682460 100644
--- a/tensorflow_hub/compressed_module_resolver.py
+++ b/tensorflow_hub/compressed_module_resolver.py
@@ -59,7 +59,10 @@ def _append_compressed_format_query(handle):
   parsed = list(urlparse.urlparse(handle))
   qsl = urlparse.parse_qsl(parsed[4])
   qsl.append(_COMPRESSED_FORMAT_QUERY)
-  parsed[4] = urlencode(qsl)
+  # NOTE: Cast to string to avoid urlunparse to deal with mixed types.
+  # This happens due to backport of urllib.parse into python2 returning an
+  # instance of <class 'future.types.newstr.newstr'>.
+  parsed[4] = str(urlencode(qsl))
   return urlparse.urlunparse(parsed)
 
 
